controllers/rack.go

```diff
@@ -540,6 +540,12 @@ func (r *SingleClusterReconciler) upgradeRack(
 
 		podNames := getPodNames(podsBatch)
 
+		// For each block volume removed from a namespace, pod status dirtyVolumes is appended with that volume name.
```

Comment:
I think we should put this function inside deletePodAndEnsureImageUpdated function, which is called after checking for migrations, quiesce.

Reply:
done